OrbitVulkanLayer/Main.cpp

```diff
@@ -0,0 +1,270 @@
+}
+
+VKAPI_ATTR VkResult VKAPI_CALL OrbitQueuePresentKHR(VkQueue queue,
+                                                    const VkPresentInfoKHR* present_info) {
+  return controller_.OnQueuePresentKHR(queue, present_info);
+}
+
+VKAPI_ATTR void VKAPI_CALL OrbitCmdBeginDebugUtilsLabelEXT(VkCommandBuffer command_buffer,
+                                                           const VkDebugUtilsLabelEXT* label_info) {
+  controller_.OnCmdBeginDebugUtilsLabelEXT(command_buffer, label_info);
+}
+
+VKAPI_ATTR void VKAPI_CALL OrbitCmdEndDebugUtilsLabelEXT(VkCommandBuffer command_buffer) {
+  controller_.OnCmdEndDebugUtilsLabelEXT(command_buffer);
+}
+
+VKAPI_ATTR void VKAPI_CALL OrbitCmdDebugMarkerBeginEXT(
+    VkCommandBuffer command_buffer, const VkDebugMarkerMarkerInfoEXT* marker_info) {
+  controller_.OnCmdDebugMarkerBeginEXT(command_buffer, marker_info);
+}
+
+VKAPI_ATTR void VKAPI_CALL OrbitCmdDebugMarkerEndEXT(VkCommandBuffer command_buffer) {
+  controller_.OnCmdDebugMarkerEndEXT(command_buffer);
+}
+
+// ----------------------------------------------------------------------------
+// Layer enumeration functions
+// ----------------------------------------------------------------------------
+
+VKAPI_ATTR VkResult VKAPI_CALL
+OrbitEnumerateInstanceLayerProperties(uint32_t* property_count, VkLayerProperties* properties) {
+  return controller_.OnEnumerateInstanceLayerProperties(property_count, properties);
+}
+
+// Deprecated by Khronos, but we'll support it in case older applications still
+// use it.
+VKAPI_ATTR VkResult VKAPI_CALL OrbitEnumerateDeviceLayerProperties(
+    VkPhysicalDevice /*physical_device*/, uint32_t* property_count, VkLayerProperties* properties) {
+  // This function is supposed to return the same results as
+  // EnumerateInstanceLayerProperties since device layers were deprecated.
+  return controller_.OnEnumerateInstanceLayerProperties(property_count, properties);
+}
+
+VKAPI_ATTR VkResult VKAPI_CALL OrbitEnumerateInstanceExtensionProperties(
+    const char* layer_name, uint32_t* property_count, VkExtensionProperties* properties) {
+  return controller_.OnEnumerateInstanceExtensionProperties(layer_name, property_count, properties);
+}
+
+VKAPI_ATTR VkResult VKAPI_CALL OrbitEnumerateDeviceExtensionProperties(
+    VkPhysicalDevice physical_device, const char* layer_name, uint32_t* property_count,
+    VkExtensionProperties* properties) {
+  return controller_.OnEnumerateDeviceExtensionProperties(physical_device, layer_name,
+                                                          property_count, properties);
+}
+
+// ----------------------------------------------------------------------------
+// GetProcAddr functions
+// ----------------------------------------------------------------------------
+
+#define ORBIT_GETPROCADDR(func)                              \
+  if (strcmp(name, "vk" #func) == 0) {                       \
+    return absl::bit_cast<PFN_vkVoidFunction>(&Orbit##func); \
+  }
```

Comment:
If we want to keep this macro to do the `if` instead of something else like filling a data structure, I still think that it could be renamed to tell what it's doing. For the name, maybe something like "RETURN_ORBIT_FUNCTION_IF_MATCHES_VK_FUNCTION`? Ugly but at least more descriptive. Better ideas are for sure possible.

Also, I would personally wrap it in a `do {} while(false)` so that it requires to be followed by a semicolon.

Reply:
Agree.